src/main/java/io/r2dbc/postgresql/SSLConfig.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2017-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.r2dbc.postgresql;
+
+
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import io.r2dbc.postgresql.util.Assert;
+import reactor.netty.tcp.SslProvider;
+import reactor.util.annotation.Nullable;
+
+import javax.net.ssl.HostnameVerifier;
+import java.io.File;
+import java.util.function.Supplier;
+
+import static reactor.netty.tcp.SslProvider.DefaultConfigurationType.TCP;
+
+public class SSLConfig {
```

Comment:
Let's move that class into the `client` package to avoid package cycles and make it `final`.

Reply:
The configuration class getters has package-private access, we should change it to public then.

Reply:
What class outside of the client package is calling the getters?

Reply:
SSLConfig in static factory method.

Reply:
What if we move `createSslConfig` into `PostgresqlConnectionConfiguration` so the configuration class can create  `SslConfig`?